fastlane/lanes/localization.rb

```diff
@@ -95,78 +94,36 @@
-  # This lane gets the data from the txt files in the `WordPress/metadata/` folder
-  # and updates the `.po` file that is then picked by GlotPress for translations.
-  # -----------------------------------------------------------------------------------
-  # Usage:
-  # fastlane update_wordpress_appstore_strings [version:<version>]
-  #
-  # Example:
-  # fastlane update_wordpress_appstore_strings version:10.3
-  #####################################################################################
-  desc 'Updates the PlayStoreStrings.po file for WordPress'
-  lane :update_wordpress_appstore_strings do |options|
-    metadata_folder = File.join(Dir.pwd, '..', 'WordPress', 'metadata')
-    version = options.fetch(:version, android_get_app_version)
-
-    # <key in po file> => <path to txt file to read the content from>
-    files = {
-      release_note: File.join(metadata_folder, 'release_notes.txt'),
-      release_note_short: File.join(metadata_folder, 'release_notes_short.txt'),
-      play_store_app_title: File.join(metadata_folder, 'title.txt'),
-      play_store_promo: File.join(metadata_folder, 'short_description.txt'),
-      play_store_desc: File.join(metadata_folder, 'full_description.txt')
-    }
-    files.merge!((1..9).map do |n|
-      [:"play_store_screenshot_#{n}", File.join(metadata_folder, "screenshot_#{n}.txt")]
-    end.to_h)
-
-    update_po_file_for_metadata_localization(
-      po_path: File.join(metadata_folder, 'PlayStoreStrings.po'),
-      sources: files,
-      release_version: version,
-      commit_message: "Update WordPress `PlayStoreStrings.po` for version #{version}"
-    )
-  end
-
-  #####################################################################################
-  # update_jetpack_appstore_strings
-  # -----------------------------------------------------------------------------------
-  # This lane gets the data from the txt files in the `WordPress/jetpack_metadata/` folder
-  # and updates the `.po` file that is then picked by GlotPress for translations.
-  # -----------------------------------------------------------------------------------
-  # Usage:
-  # fastlane update_jetpack_appstore_strings [version:<version>]
-  #
-  # Example:
-  # fastlane update_jetpack_appstore_strings version:10.3
-  #####################################################################################
-  desc 'Updates the PlayStoreStrings.po file for Jetpack'
-  lane :update_jetpack_appstore_strings do |options|
-    metadata_folder = File.join(Dir.pwd, '..', 'WordPress', 'jetpack_metadata')
-    version = options.fetch(:version, android_get_app_version)
+    # If no `app:` is specified, call this for both WordPress and Jetpack
+    apps = options[:app].nil? ? %i[wordpress jetpack] : Array(options[:app]&.downcase&.to_sym)
+
+    apps.each do |app|
+      app_values = APP_SPECIFIC_VALUES[app]
+
+      metadata_folder = File.join(PROJECT_ROOT_FOLDER, 'WordPress', app_values[:metadata_dir])
```

Comment:
Suggestion (💡): I find lots of references to this 'WordPress' folder here and there, and it took me some seconds to make sure that this code is not doing something specific for the WordPress vs. the Jetpack case, thus maybe it it time we extract this constant to some naming convention that doesn't reference `WordPress`, but applies to `Jetpack` as well, something like APP_MODULE_FOLDER = 'WordPress'. Wdyt? 🤔

PS.1: I am also seeing lot of lowercased constants like 'wordpress' and `jetpack`, for the `app` and `app_flavor` parameters. If you like my above suggestion I would also make an additional effort now and extract those into constants as well, just so to make those explicit and possibly in the future more mainstream, like:
- `app`
    - WORDPRESS_APP_NAME = 'wordpress'
    - JETPACK_APP_NAME = 'jetpack'
- `app_flavor`
    - WORDPRESS_APP_FLAVOR = 'wordpress`
    - JETPACK_APP_FLAVOR = 'jetpack'

Maybe adding the above to the `APP_SPECIFIC_VALUES` values is the right way, or reuse them from there, or somewhere else, I am not sure, but I would love to not have it there as is, like a magic number not being explicitly extracted to a constant... 🤔 

PS.2: All these suggestions apply to all such fastlane files, not just `localization.rb`, but we can start with that if extracting for all is too out-of-scope for this PR. 🤔

Reply:
> Suggestion (💡): I find lots of references to this 'WordPress' folder here and there, and it took me some seconds to make sure that this code is not doing something specific for the WordPress vs. the Jetpack case, thus maybe it it time we extract this constant to some naming convention that doesn't reference WordPress, but applies to Jetpack as well, something like APP_MODULE_FOLDER = 'WordPress'. Wdyt?

Note that this `WordPress/` folder is referencing the folder at the root of the repo. We don't have a `Jetpack` folder at the root of the repo, just a `WordPress/` one (and then we have subfolders in `src` for `main` vs `wordpress` vs `jetpack` for the different flavors).

I'm totally ok to extract that `WordPress` root folder name into some constant like `APP_MODULE_FOLDER`, though. 

I'm also OK in adding an `app_flavor` entry to the `APP_SPECIFIC_VALUES` hash, to separate the name of the key used (which was based on the app flavor name to begin with, because we had to choose something to base the `app:` keys off of 😛 … but we can totally make the name of the app flavor explicit as an `app_flavor` entry to make the key used for `APP_SPECIFIC_VALUES` and the app flavor disconnected)

I plan to do that in a separate PR though, in fact I already [have a WIP branch that cleans up some of those constants and the `APP_SPECIFIC_VALUES`](https://github.com/wordpress-mobile/WordPress-Android/compare/tooling/screenshots...tooling/screenshots-lane-tweaks) that I plan to improve further 🙂

Reply:
> Note that this WordPress/ folder is referencing the folder at the root of the repo. We don't have a Jetpack folder at the root of the repo, just a WordPress/ one (and then we have subfolders in src for main vs wordpress vs jetpack for the different flavors).

Yeap, I know, I understand that to 100%, but this is just unfortunate because `Jetpack` was born later and reused the `WordPress` codebase! 👍 

> I'm totally ok to extract that WordPress root folder name into some constant like APP_MODULE_FOLDER, though.

Exactly, thanks, having to refer to a `module` instead of `WordPress` seems a much better choice.

> I'm also OK in adding an app_flavor entry to the APP_SPECIFIC_VALUES hash, to separate the name of the key used (which was based on the app flavor name to begin with, because we had to choose something to base the app: keys off of 😛 … but we can totally make the name of the app flavor explicit as an app_flavor entry to make the key used for APP_SPECIFIC_VALUES and the app flavor disconnected)

👍 

> I plan to do that in a separate PR though, in fact I already [have a WIP branch that cleans up some of those constants and the APP_SPECIFIC_VALUES](https://github.com/wordpress-mobile/WordPress-Android/compare/tooling/screenshots...tooling/screenshots-lane-tweaks) that I plan to improve further 🙂

Coolio, thanks for even considering! 🥇